src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs

```diff
@@ -523,7 +523,21 @@ private static void SanityCheckBlob(object obj, string[] blobs)
             }
         }
 
-        private static int GetBlobIndex(int length) => PlatformDetection.IsFullFramework ? 1 : (length == 3 ? 2 : 0);
+        private static TargetFrameworkMonikers GetTargetFrameworkMonikers() => PlatformDetection.IsFullFramework ? TargetFrameworkMonikers.NetFramework : TargetFrameworkMonikers.Netcoreapp;
```

Comment:
This function can be used to get the target framework...when we will have apis for detecting mono or versions of netcorapp. and can update the respective Index accordingly.   
 (No swapping required).